button-examples/Android/android-button-example.md

```diff
@@ -23,213 +21,387 @@ There are four types of buttons:
+| **Padding** (space between icon and text label) | `app:iconPadding` | `setIconPadding`<br/>`getIconPadding` | `4dp` |
+
+</details>
+
+See the full list of [attrs](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/button/res/values/attrs.xml).
+
+</details>
+
+<details>
+<summary><b>Styles</b></summary>
+<br>
+
+|  | Style |
+| --- | --- |
+| **Default style** | `Widget.MaterialComponents.Button.TextButton` |
+| **Icon style** | `Widget.MaterialComponents.Button.TextButton.Icon` |
+
+See the full list of [styles](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/button/res/values/styles.xml).
+
+Default style theme attribute: N/A
+
+</details>
 
 ## Outlined button
 
 [Outlined buttons](https://material.io/components/buttons/#outlined-button) are medium-emphasis buttons. They contain actions that are important, but aren’t the primary action in an app.
 
+### Outlined button examples
 
-### Outlined button example
-
-Source code API:
+API and source code:
 
 * `MaterialButton`
     * [Class description](https://developer.android.com/reference/com/google/android/material/button/MaterialButton)
     * [GitHub source](https://github.com/material-components/material-components-android/blob/master/lib/java/com/google/android/material/button/MaterialButton.java)
     
 The following example shows an outlined button with a text label and stroked container.    
 
-!["Outlined button example in Android with purple text surrounded by a gray outline"](assets/outlined-button.svg)
+!["Outlined button"](assets/outlined-button.svg)
 
+In the layout:
 ```xml
-    <Button
-        android:id="@+id/outlinedButton"
-        style="@style/Widget.MaterialComponents.Button.OutlinedButton"
-        android:layout_width="wrap_content"
-        android:layout_height="wrap_content"
-        android:text="Outlined button"
-        ...
-    />
+<Button
+    android:id="@+id/outlinedButton"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:text="Outlined button"
+    style="?attr/materialButtonOutlinedStyle"
+/>
+```
+
+_**Note:** `<Button>` is auto-inflated as `<com.google.android.material.button.MaterialButton>` via `MaterialComponentsViewInflater` when using a non-Bridge `Theme.MaterialComponents.*` theme._
+
+In code:
+```kt
+outlinedButton.setOnClickListener {
+    // Respond to button press
+}
+```
+
+The following example shows an outlined button with an icon.
+
+!["Outlined button with icon"](assets/outlined-button-icon.svg)
+
+In the layout:
+```xml
+<Button
+    ...
+    app:icon="@drawable/ic_add_24dp"
+    style="@style/Widget.MaterialComponents.Button.OutlinedButton.Icon"
+/>
 ```
 
 ### Anatomy and key properties
 
-An outline button has text, a container, and an optional icon.
+An outlined button has a text label, a stroked container and an optional icon.
```

Comment:
I will forever hold to the Oxford comma.

"I'd like to thank my parents, Ayn Rand and God."